lib/DriverTool/autolink_extract_main.cpp

```diff
@@ -140,8 +142,15 @@ extractLinkerFlagsFromObjectFile(const llvm::object::ObjectFile *ObjectFile,
       llvm::SmallVector<llvm::StringRef, 4> SplitFlags;
       SectionData->split(SplitFlags, llvm::StringRef("\0", 1), -1,
                          /*KeepEmpty=*/false);
-      for (const auto &Flag : SplitFlags)
-        LinkerFlags.push_back(Flag.str());
+      for (const auto &Flag : SplitFlags) {
+        // If this is a library '-lxxx' flag, only add it if we have not seen it before
```

Comment:
I worry about this approach because you have to consider the following equivalences:
1. `-l:library.a`
2. `-static` `-lrary`
3. `lib/library.a`

and the following:
1. `-l:library.so`
2. `-shared` `-lrary`
3. `lib/liblibrary.so`

Furthermore, the order of the libraries is relevant to linking.  The conversion to an unordered_set breaks linking semantics - order of library specification on the command line indicates how to resolve symbols.  This can change the symbolic resolution.